File: Pset_1/Cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>


int main(void)
{
    // TODO: Prompt for input
    float input;
    do
    {
        input = get_float("Change owed in dollars: ");
    }
    while (input < 0);

    // TODO: round to cents
    int cents = round(input * 100);


    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;


    // TODO: Calculate
    int sum = 0;


    while (cents > 0)
    {
        if (cents >= quarter)
        {
            cents = cents - quarter;
            sum++;
        }
        else if (cents >= dime)
        {
            cents = cents - dime;
            sum++;
        }
        else if (cents >= nickel)
        {
            cents = cents - nickel;
            sum++;
        }
        else if (cents >= penny)
        {
            cents = cents - penny;
            sum++;
        }
        // printf("%i", cents);
        // printf("\n");
    }
    printf("Left1: %i\n", sum);

}

File: Pset_6/Cash/cash.py
from cs50 import get_float

# Coins
quarter = 25
dime = 10
nickel = 5
penny = 1


# Empty list of coins
coins = 0


while True:

    try:
        # Get input from the user
        userInput = get_float("Change owed in dollars: ")
        if userInput >= 0:
            break

    except ValueError:
        # Ask again for an input - continue returns the control to the beginning of the while loop
        continue


# Round to cents
cents = round(userInput * 100)


# Calculation and deduction
while cents > 0:
    if cents >= quarter:
        cents -= quarter
        coins += 1
    elif cents >= dime:
        cents -= dime
        coins += 1
    elif cents >= nickel:
        cents -= nickel
        coins += 1
    elif cents >= penny:
        cents -= penny
        coins += 1


print(coins)